### PR TITLE
CUSTCOM-14 Improvements in stop-domain process

### PR DESCRIPTION
**Pull request by MeroRai**
# Description
This a feature to improve the stop-domain process. Now check is performed to see if the hostname passed by the `--host` parameter represents the current machine. 

# Testing

### Testing Performed
Ran the following command to test:
```
./asadmin start-domain
./asadmin --host $MY_HOSTNAME --port 4848 stop-domain
```

### Testing Environment
Zulu JDK 1.8_222 on Elementary OS 0.4.1 Loki with Maven 3.5.4

# Documentation
Not required

### CODE REVIEW

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java [hunk 1]
```diff
@@ -222,18 +220,10 @@ public synchronized void start() {
             // See issue #5596 to know why we set context CL as common CL.
             Thread.currentThread().setContextClassLoader(
                     commonCLS.getCommonClassLoader());
-            if (state.compareAndSet(State.INITIAL, State.STARTING) || state.compareAndSet(State.SHUT_DOWN, State.STARTING)) {
```
Comment (MattGill98) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java:225]: This seems like a stronger way of checking the startup state of the server - why has this been removed?

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java [hunk 2]
```diff
@@ -475,23 +466,31 @@ public static void printModuleStatus(ModulesRegistry registry, Level level) {
         logger.log(level, sb.toString());
     }
 
+    // TODO(Sahoo): Revisit this method after discussing with Jerome.
```
Comment (MattGill98) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java:469]: Who's Jerome? 😂

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java (hunk 1 above)
Comment (MeroRai) [reply nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java:225]: I messed up my commit history and mistakenly removed the previous commit made by Patrick. I will fix it and update the PR. Thanks for pointing out

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java (hunk 2 above)
Comment (MeroRai) [reply nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java:469]: Read the above comment

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java [hunk 3]
```diff
@@ -543,9 +543,11 @@ public synchronized void stop() {
...
     /**
-     * Proceed to the given run level.
+     * Proceed to the given run level using the given {@link AppServerActivator}.
      *
      * @param runLevel   the run level to proceed to
+     * @param activator  an {@link AppServerActivator activator} used to
+     *                   activate/deactivate the services
      * @return false if an error occurred that required server shutdown; true otherwise
      */
     private boolean proceedTo(int runLevel) {
```
Comment (svendiedrichsen) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/AppServerStartup.java:553]: Is there a parameter AppServerActivator missing? At least you have added the comment about one.

**Comment by MeroRai** [comment]
jenkins test please

**Comment by MeroRai** [comment]
jenkins test please

**Comment by MeroRai** [comment]
jenkins test please

**Pull request merged**